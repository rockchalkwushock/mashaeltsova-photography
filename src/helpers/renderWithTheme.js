import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
