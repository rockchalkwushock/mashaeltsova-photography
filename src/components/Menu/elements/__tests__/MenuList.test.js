import React from 'react'

import { renderWithTheme } from '../../../../helpers'
import { MenuList } from '../index'

describe('Styled Component: <MenuList />', () => {
  let wrapper
  beforeAll(() => {
    wrapper = renderWithTheme(<MenuList />)
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('display', 'grid')
    expect(wrapper).toHaveStyleRule('grid-gap', '10px')
    expect(wrapper).toHaveStyleRule('grid-template-columns', 'repeat(1fr)')
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('margin-top', '10px')
    expect(wrapper).toHaveStyleRule('text-align', 'center')
  })
})
