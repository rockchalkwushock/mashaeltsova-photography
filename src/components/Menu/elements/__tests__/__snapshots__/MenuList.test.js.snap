// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <MenuList /> should render markup 1`] = `
.c0 {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(1fr);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 10px;
  text-align: center;
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <MenuList>
    <ul
      className="c0"
    />
  </MenuList>
</ThemeProvider>
`;
